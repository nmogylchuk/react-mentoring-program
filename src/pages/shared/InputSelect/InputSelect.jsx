import React from "react";
import PropTypes from "prop-types";
import 'pages/shared/InputSelect/InputSelect.scss';

function InputSelect({label, name, genres}) {
    return (
        <div className="input-select">
            <label className="input-select__label">{label}</label>
                <select className="input-select__list" name={name}>
                    {Object.entries(genres).map(([key, genre]) => (
                        <option className="input-select__item" key={key}>{genre}</option>
                    ))}
                </select>
        </div>
    )
}

InputSelect.propTypes = {
    label: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    genres: PropTypes.array.isRequired
};

export default InputSelect;
