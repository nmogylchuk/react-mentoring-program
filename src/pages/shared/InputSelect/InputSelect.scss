@import 'src/assets/styles/variables';

.input-select {
  margin-top: 5px;
  display: flex;
  flex-direction: column;

  &__label {
    color: $red;
    font-size: 1.2rem;
    text-transform: uppercase;
  }

  &__list {
    display: inline-block;
    min-height: 30px;
    border: none;
    color: $white;
    font-size: 1.6rem;
    background: $grey;
    padding: 10px 20px;
    margin: 10px 0;
    width: 100%;

  }
}
