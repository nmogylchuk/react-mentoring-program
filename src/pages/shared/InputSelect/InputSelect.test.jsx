import React from 'react';
import rerender from 'react-test-renderer';
import InputSelect from "./InputSelect";
import {jest} from "@jest/globals";

describe('when Input Select', () => {

    const movie = {
        "id": 447365,
        "title": "Guardians of the Galaxy Vol. 3",
        "tagline": "",
        "vote_average": 0,
        "vote_count": 9,
        "release_date": "2020-05-01",
        "poster_path": "https://image.tmdb.org/t/p/w500/ldoY4fTZkGISMidNw60GHoNdgP8.jpg",
        "overview": "The third film based on Marvel's Guardians of the Galaxy.",
        "budget": 0,
        "revenue": 0,
        "genres": [
            "Action",
            "Adventure",
            "Science Fiction"
        ],
        "runtime": null
    }

    test('then snapshot created', () => {
        const inputSelectComponent = rerender.create(<InputSelect label="Genres" name="genres" genres={movie.genres}
                                                                  value={movie.genres} onChange={jest.fn()}
        />);
        const tree = inputSelectComponent.toJSON();
        expect(tree).toMatchSnapshot();
    });
});