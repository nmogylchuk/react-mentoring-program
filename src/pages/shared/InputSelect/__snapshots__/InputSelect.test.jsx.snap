// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when Input Select then snapshot created 1`] = `
<div
  className="input-select"
>
  <label
    className="input-select__label"
  >
    Genres
  </label>
  <select
    className="input-select__list"
    multiple={true}
    name="genres"
    value={
      Array [
        "Action",
        "Adventure",
        "Science Fiction",
      ]
    }
  >
    <option
      className="input-select__item"
    >
      Action
    </option>
    <option
      className="input-select__item"
    >
      Adventure
    </option>
    <option
      className="input-select__item"
    >
      Science Fiction
    </option>
  </select>
</div>
`;
